lib/jxl/jpeg/dec_jpeg_data_writer.cc

```diff
@@ -783,6 +872,9 @@ SerializationStatus JXL_NOINLINE DoEncodeScan(const JPEGData& jpg,
         ss.restarts_to_go = restart_interval;
         memset(ss.last_dc_coeff, 0, sizeof(ss.last_dc_coeff));
       }
+      // compressed size per block cannot be more than 256 bytes per component
+      Reserve(bw, 256 * scan_info.num_components);
```

Comment:
Right, will change it to blocks-per-MCU.

The 256 bytes per block bound I took from libjpeg-turbo. The worst-case scenario you describe would happen if the huffman codes and extra bits would both be all-1s, right? So I guess you're right, in theory it could be 512 bytes.

I don't see a convenient already-available variable for blocks-per-MCU so I'll just do one `Reserve` per component instead, it will not make much of a difference. It's just good to get the `Reserve` out of the inner loop (it used to be reserving only 12 bytes at a time).